### PR TITLE
QOL-7857 Extract reused steps into custom recipes

### PR DESCRIPTION
**Pull request by ThrawnCA**

### CODE REVIEW

File: recipes/ckanweb-deploy-exts.rb [hunk 1]
```diff
@@ -109,91 +109,25 @@
...
-			fi
-			if [ -f "$REQUIREMENTS_FILE" ]; then
-				#{pip} install -r $REQUIREMENTS_FILE
-			fi
-			if [ -f "pip-requirements.txt" ]; then
-				#{pip} install -r "pip-requirements.txt"
-			fi
-		EOS
-	end
-
```
Comment (duttonw) [inline recipes/ckanweb-deploy-exts.rb:196]: is the ckan config.ini building in batch also?

File: recipes/ckanweb-efs-setup.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 # Update EFS Data directory for CKAN
 #
-include_recipe "datashades::httpd-efs-setup"
+include_recipe "datashades::efs-setup"
```
Comment (duttonw) [inline recipes/ckanweb-efs-setup.rb:20]: does this not break things?

File: recipes/ckanweb-efs-setup.rb [hunk 3]
```diff
@@ -56,33 +56,18 @@
 
 if extra_disk_present then
     real_log_dir = "#{extra_disk}/#{service_name}"
+
+    datashades_move_and_link(var_log_dir) do
```
Comment (duttonw) [inline recipes/ckanweb-efs-setup.rb:60]: nice use of a function :+1:

File: recipes/ckanweb-efs-setup.rb (hunk 2 above)
Comment (duttonw) [reply recipes/ckanweb-efs-setup.rb:20]: what happened to using httpd-efs-setup.rb ?

**Review by duttonw — COMMENTED** [review]
i'm uncertain if this would work.

File: recipes/ckanweb-deploy-exts.rb (hunk 1 above)
Comment (ThrawnCA) [reply recipes/ckanweb-deploy-exts.rb:196]: That happens in `ckan-deploy.rb`, which runs on both layers.

File: recipes/ckanweb-efs-setup.rb (hunk 2 above)
Comment (ThrawnCA) [reply recipes/ckanweb-efs-setup.rb:20]: It's now on `ckanweb-setup.rb` line 28. So it will run on the web layer, but not batch.

**Review by duttonw — APPROVED** [review]

**Pull request merged**